Automattic/jetpack
# Latest Instagram Posts: Support for Multiple User Accounts

![https://i.imgur.com/I6GoNjB.gif](https://i.imgur.com/I6GoNjB.gif)

Fixes #15848

#### Changes proposed in this Pull Request:

* Add support for multiple user accounts in the Latest Instagram Posts block.
  * Users with active Instagram tokens, can reuse them straight away.
  * Users without, will need to connect to Instagram through the usual popup.
* Disconnecting the block won't delete the token anymore, but simply remove it from the block attributes.
  * To permanently delete the token, users will need to head to Marketing/Connections in Calypso, or do it directly from Instagram.
* Remove 2 unnecessary endpoints (`GET /access-token` and `DELETE /delete-access-token`), and corresponding helper code.

#### Screenshots

cc @davemart-in and @olesyabrk for suggestions about the placeholder copy, especially the account selector, and the disconnection button in the block sidebar.

User without connected Instagram accounts:
<img width="629" alt="Screenshot 2020-05-19 at 19 28 26" src="https://user-images.githubusercontent.com/2070010/82365720-efb12000-9a08-11ea-8f56-d549202bc650.png">

User with connected Instagram accounts:
<img width="626" alt="Screenshot 2020-05-20 at 11 48 30" src="https://user-images.githubusercontent.com/2070010/82438004-457cdb00-9a90-11ea-8ef6-2357ea7a6360.png">

Disconnection button:
<img width="275" alt="Screenshot 2020-05-20 at 11 48 23" src="https://user-images.githubusercontent.com/2070010/82438308-bc19d880-9a90-11ea-9199-a2f398c140ee.png">

#### Testing instructions:

WPCOM prerequisites: apply D43668-code and D43669-code (hopefully it patches cleanly).

- Smoke test the Latest Instagram Posts block, and make sure the UX feels good.
- Try connecting more than one Instagram account.
- Make sure connecting new accounts is reflected in the account selector.
- Make sure that after disconnecting a block, the previous account is automatically selected in the account selector.
- Make sure blocks connected to different accounts are visible on the front end to visitors.

#### Proposed changelog entry for your changes:

* N/A
